### PR TITLE
Fixing training/validation accuracy to match in progress table and state in C-API of Activity Classifier

### LINKED ISSUE #2123: C-API of Activity Classifier accuracy from progress table does not match the one stored in the state
The C-API of Activity Classifier calculates the training and validation accuracies in the progress table ```per_row```. The accuracies stored in the state were calculated using evaluate function which was always ```per_row```. Ideally, the validation accuracy of the last iteration as well as the validation accuracy stored in the state must match. There is a slight mismatch which should be investigated. 

```model2.train(data2, target='activity',session_id='exp_id',validation_data='auto',options={'prediction_window' :50, 'max_iterations ' : 20})
Pre-processing 672978 samples...
Using sequences of size 1000 for model creation.
Processed a total of 55 sessions.
Using GPU to create model (AMD Radeon Pro 560X)
+--------------+----------------+--------------+---------------------+-----------------+--------------+
| Iteration    | Train Accuracy | Train Loss   | Validation Accuracy | Validation Loss | Elapsed Time |
+--------------+----------------+--------------+---------------------+-----------------+--------------+
| 1            | 0.650028       | 0.901889     | 0.867175            | 0.941983        | 3.07s        |
| 2            | 0.825928       | 0.482736     | 0.88453             | 0.380414        | 3.88s        |
| 3            | 0.86012        | 0.382338     | 0.896773            | 0.268208        | 4.67s        |
| 4            | 0.875827       | 0.329919     | 0.913529            | 0.223856        | 5.47s        |
| 5            | 0.887411       | 0.293027     | 0.923172            | 0.220975        | 6.26s        |
| 6            | 0.893232       | 0.271588     | 0.936401            | 0.197999        | 7.08s        |
| 11           | 0.921169       | 0.187454     | 0.932994            | 0.151633        | 11.10s       |
| 16           | 0.936732       | 0.140357     | 0.933897            | 0.144817        | 15.29s       |
| 20           | 0.940048       | 0.125558     | 0.935484            | 0.156982        | 18.71s       |
+--------------+----------------+--------------+---------------------+-----------------+--------------+
Validation Accuracy = 0.936469```

### PR DESCRIPTION
**Pull request by shreyajain17**
- fixes #2123

-  The C-API of Activity Classifier calculates the training and validation accuracies in the progress table per_row and the accuracies stored in the state were calculated using evaluate function which was always per_row. Ideally, the validation accuracy of the last iteration as well as the validation accuracy stored in the state must match. There was a slight mismatch which is now resolved. 

```
model2.train(data, target='label',session_id='session_id',validation_data=val,options={'prediction_wi
   ...: ndow' :100, 'max_iterations' : 10})
Pre-processing 269134 samples...
Using sequences of size 2000 for model creation.
Processed a total of 1231 sessions.
Using GPU to create model (AMD Radeon Pro 560X)
+--------------+----------------+--------------+---------------------+-----------------+--------------+
| Iteration    | Train Accuracy | Train Loss   | Validation Accuracy | Validation Loss | Elapsed Time |
+--------------+----------------+--------------+---------------------+-----------------+--------------+
| 1            | 0.417677       | 0.416866     | 0.586352            | 0.218555        | 7.20s        |
| 2            | 0.402075       | 0.339111     | 0.642626            | 0.13199         | 9.32s        |
| 3            | 0.495712       | 0.173582     | 0.716688            | 0.098844        | 11.44s       |
| 4            | 0.803262       | 0.106824     | 0.86457             | 0.0626355       | 13.55s       |
| 5            | 0.871046       | 0.060315     | 0.919874            | 0.046532        | 15.66s       |
| 6            | 0.937284       | 0.0360295    | 0.959411            | 0.0355047       | 17.78s       |
| 10           | 0.991082       | 0.00651922   | 0.844033            | 0.0348628       | 26.31s       |
+--------------+----------------+--------------+---------------------+-----------------+--------------+
Validation Accuracy : 0.844033
In [6]: model2.train(data, target='label',session_id='session_id',validation_data=val,options={'prediction_wi
   ...: ndow' :100, 'max_iterations' : 10})
Pre-processing 269134 samples...
Using sequences of size 2000 for model creation.
Processed a total of 1231 sessions.
Using GPU to create model (AMD Radeon Pro 560X)
+--------------+----------------+--------------+---------------------+-----------------+--------------+
| Iteration    | Train Accuracy | Train Loss   | Validation Accuracy | Validation Loss | Elapsed Time |
+--------------+----------------+--------------+---------------------+-----------------+--------------+
| 1            | 0.38466        | 0.466429     | 0.498221            | 0.184025        | 6.79s        |
| 2            | 0.405359       | 0.343032     | 0.615055            | 0.137465        | 8.95s        |
| 3            | 0.503634       | 0.20884      | 0.732455            | 0.0988752       | 11.07s       |
| 4            | 0.73087        | 0.111564     | 0.836675            | 0.0574139       | 13.21s       |
| 5            | 0.847084       | 0.0690134    | 0.871766            | 0.0467798       | 15.33s       |
| 6            | 0.958288       | 0.0274774    | 0.94979             | 0.0305351       | 17.48s       |
| 10           | 0.993483       | 0.00509692   | 0.992238            | 0.0164606       | 26.01s       |
+--------------+----------------+--------------+---------------------+-----------------+--------------+
Validation Accuracy : 0.992238
```

### CODE REVIEW

File: src/toolkits/activity_classification/activity_classifier.cpp [hunk 1]
```diff
@@ -859,8 +864,12 @@ void activity_classifier::perform_training_iteration() {
   const size_t iteration_idx = read_state<flex_int>("training_iterations");
 
   float cumulative_batch_loss = 0.f;
-  float cumulative_batch_accuracy = 0.f;
+  // float cumulative_batch_accuracy = 0.f;
```
Comment (nickjong) [inline src/toolkits/activity_classification/activity_classifier.cpp]: Don't need this anymore?

File: src/toolkits/activity_classification/activity_classifier.cpp [hunk 2]
```diff
@@ -859,8 +864,12 @@ void activity_classifier::perform_training_iteration() {
...
   size_t num_batches = 0;
+  size_t cumulative_train_chunk = 0;
+  size_t cumulative_train_chunks = 0;
+  size_t num_train_samples = 0;
+  size_t total_train_samples = 0;
```
Comment (nickjong) [inline src/toolkits/activity_classification/activity_classifier.cpp]: Same as above, let's move two of these into the lambda. And maybe `{train/batch}_num_{correct/samples}`

File: src/toolkits/activity_classification/activity_classifier.cpp [hunk 3]
```diff
@@ -245,15 +243,18 @@ std::tuple<float, float> activity_classifier::compute_validation_metrics(
 
     while (pending_batches.size() > remaining) {
 
+      size_t batch_num_correct = 0;
```
Comment (nickjong) [inline src/toolkits/activity_classification/activity_classifier.cpp]: Nit: Let's improve the spatial locality here by moving these variable definitions to just before they're used

File: src/toolkits/activity_classification/activity_classifier.cpp [hunk 4]
```diff
@@ -864,12 +866,11 @@ void activity_classifier::perform_training_iteration() {
   const size_t iteration_idx = read_state<flex_int>("training_iterations");
 
   float cumulative_batch_loss = 0.f;
-  // float cumulative_batch_accuracy = 0.f;
   size_t num_batches = 0;
-  size_t cumulative_train_chunk = 0;
-  size_t cumulative_train_chunks = 0;
-  size_t num_train_samples = 0;
-  size_t total_train_samples = 0;
+  size_t batch_num_correct = 0;
```
Comment (nickjong) [inline src/toolkits/activity_classification/activity_classifier.cpp]: Let's move `batch_num_correct` and `batch_num_samples` into the lambda below
